vendure-ecommerce/vendure
# feat(core): Enable to change the name of the cookie for Admin API

### Issue

Fixes #2481 

### Description

In this PR, I added a new `authOptions.adminCookieName` optional parameter in the Vendure config. 
It makes it possible to give a custom name to the token set for the Admin API than the one given to the Shop API.

For now, here is the cookie naming behavior:
- If not setting the `authOptions.cookieOptions.name` parameter:
  - Both the Admin API and Shop API cookies names will be the default one: `'session'`
- If setting the `authOptions.cookieOptions.name` parameter to something like `'cookie-name'`:
  - Both the Admin API and Shop API cookies names will be the configured one: `'cookie-name'`

There is currently no way to give a custom name to the Admin API cookies, different from the Shop API ones.

With this changes, we can therefore set different cookies names for both APIs.

### How to test

1. In the `dev-config.ts`, change the `authOptions` as follows to name Shop API cookies `shop-api` and `shop-api.sign`, and the Admin API cookies `admin-api` and `admin-api.sig`:
   ```ts
    authOptions: {
        disableAuth: false,
        tokenMethod: ['bearer', 'cookie'] as const,
        requireVerification: true,
        customPermissions: [],
        cookieOptions: {
            name: "shop-api",
            secret: 'abc',
        },
        adminCookieName: 'admin-api'
    },
   ```
2. Start the `dev-server`
3. Open the GraphQL playground on the `/admin-api` and authenticate
4. You should see the following cookies
   ![image](https://github.com/vendure-ecommerce/vendure/assets/17927632/16e94a44-b942-4a84-99f8-e660b17257e0)
5. Open the GraphQL playground on the `/shop-api` and authenticate
6. You should now see the following cookies
   ![image](https://github.com/vendure-ecommerce/vendure/assets/17927632/438a2199-983d-4878-8e19-6824874c0362)
